pkg/controllers/subnet/subnet_controller.go

```diff
@@ -152,6 +152,9 @@ func (r *SubnetReconciler) deleteSubnetByID(subnetID string) error {
 }
 
 func (r *SubnetReconciler) deleteSubnets(nsxSubnets []*model.VpcSubnet) error {
+	if len(nsxSubnets) == 0 {
+		return nil
+	}
 	for _, nsxSubnet := range nsxSubnets {
```

Comment:
It seems that line155-157 not necessary?
If len(nsxSubnets) ==0, 171 will print subnetCount, 0

Reply:
It will print the redundant logs:
```
2024-11-05 00:05:01.936 INFO    subnetset/subnetset_controller.go:344   Successfully deleted all specified NSX Subnets  {"subnetCount": 0}
2024-11-05 00:05:01.953 INFO    subnetset/subnetset_controller.go:344   Successfully deleted all specified NSX Subnets  {"subnetCount": 0}
2024-11-05 00:05:01.971 INFO    subnetset/subnetset_controller.go:344   Successfully deleted all specified NSX Subnets  {"subnetCount": 0}
2024-11-05 00:05:01.988 INFO    subnetset/subnetset_controller.go:344   Successfully deleted all specified NSX Subnets  {"subnetCount": 0}
2024-11-05 00:05:02.002 INFO    subnetset/subnetset_controller.go:344   Successfully deleted all specified NSX Subnets  {"subnetCount": 0}
2024-11-05 00:05:02.019 INFO    subnetset/subnetset_controller.go:344   Successfully deleted all specified NSX Subnets  {"subnetCount": 0}
2024-11-05 00:05:02.034 INFO    subnetset/subnetset_controller.go:344   Successfully deleted all specified NSX Subnets  {"subnetCount": 0}
2024-11-05 00:05:02.046 INFO    subnetset/subnetset_controller.go:344   Successfully deleted all specified NSX Subnets  {"subnetCount": 0}
2024-11-05 00:05:02.079 INFO    subnetset/subnetset_controller.go:344   Successfully deleted all specified NSX Subnets  {"subnetCount": 0}
2024-11-05 00:05:02.098 INFO    subnetset/subnetset_controller.go:344   Successfully deleted all specified NSX Subnets  {"subnetCount": 0}
2024-11-05 00:05:02.113 INFO    subnetset/subnetset_controller.go:344   Successfully deleted all specified NSX Subnets  {"subnetCount": 0}
2024-11-05 00:05:02.125 INFO    subnetset/subnetset_controller.go:344   Successfully deleted all specified NSX Subnets  {"subnetCount": 0}
2024-11-05 00:05:02.144 INFO    subnetset/subnetset_controller.go:344   Successfully deleted all specified NSX Subnets  {"subnetCount": 0}
2024-11-05 00:05:02.162 INFO    subnetset/subnetset_controller.go:344   Successfully deleted all specified NSX Subnets  {"subnetCount": 0}
2024-11-05 00:05:02.189 INFO    subnetset/subnetset_controller.go:344   Successfully deleted all specified NSX Subnets  {"subnetCount": 0}
2024-11-05 00:05:02.201 INFO    subnetset/subnetset_controller.go:344   Successfully deleted all specified NSX Subnets  {"subnetCount": 0}
```

Reply:
Since GC runs every minute, a large number of Subnets will generate extensive logs.